tools/erc-repository-indexer/erc-contract-indexer/src/utils/helper.ts

```diff
@@ -91,4 +92,54 @@ export class Helper {
       }),
     };
   }
+
+  /**
+   * Merges two sorted arrays of contract objects and returns a single sorted array without duplicates.
+   * This function assumes both input arrays are sorted by `contractId` in ascending order.
+   *
+   * @param {ERCOutputInterface[]} existingContracts - The first array of contract objects, already sorted by `contractId`.
+   * @param {ERCOutputInterface[]} newContracts - The second array of contract objects, already sorted by `contractId`.
+   * @returns {ERCOutputInterface[]} A merged and sorted array of contract objects, with duplicate `contractId` entries removed.
+   */
+  static mergeAndSort(
```

Comment:
Random thought: We can explore https://lodash.com/ for array operations like merge/sort. We're using lodash in the relayer repo as far as I remember. It might be worth a simple performance benchmark against it.